### DISCUSSION TITLE
Add Trust Wallet connector

### DISCUSSION THREAD

**Discussion by albertbinance** [wagmi-dev/references#50] [Ideas]
Hello 👋

I represent the Trust Wallet team, and I'd like to start working on a connector for Trust Wallet Browser Extension. 

Please let me know if I'm good to go.

I've read you wanted to "standardize the process". Any updates on that?

**Comment by tmm** [comment]
Hi! Thanks for your interest in contributing a new Connector to the wagmi references! If you haven't already, please read the [Contributing Guidelines](https://github.com/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md#connectors). There's some additional info for adding a Connector that you likely want to check out. Next, please respond to the following questions:

1. **What **novel use-case** does the Connector provide?** A novel use-case is likely one that is not already covered by or not easily extended from another Connector (such as the `InjectedConnector` or `WalletConnectConnector`).
2. **Are the Connector's integrations production-ready and generally available?** Connectors are intended to be used by consumers in production as part of wagmi. As such, the Connector and all dependencies must be production-ready and generally available.
3. **Are you committed to actively maintaining the Connector?** The wagmi core team will provide as much assistance as possible to keep Connectors up-to-date with breaking changes from wagmi, but it is your responsibility to ensure that downstream dependencies and issues/discussions related to the Connector are handled in a timely manner.

Another thing to note is [wallets do not map directly to Connectors](https://github.com/wagmi-dev/wagmi/discussions/1346#discussioncomment-4296131) and are a higher-level concept.

**Reply by albertbinance** [reply]
Hello, and thanks for a swift response!

1. In general, we're working on improving Trust Wallet ecosystem. It includes taking care of different edge caused by having both browser extension and mobile. We also want to make sure a correct provider is retrieved when several wallets are installed, thus we don't want to only rely on injected wallet. And finally, Wagmi connector would be reused by other libraries, like rainbowkit, so it would help us not to repeat ourselves.

2. We might introduce our sdk as a new dependency, and we'll make sure it's production-ready and available on npm.

3. Yes, we'd make sure to maintain the connector to properly serve our users.

**Reply by tmm** [reply]
Appreciate your responses!

As it stands (based on https://github.com/wagmi-dev/references/pull/55), this Connector does not meet the "novel use-case" requirement since it extends an existing Connector (e.g. `InjectedConnector`) without any significant differences. Downstream consumers and libraries can use the [`InjectedConnector.getProvider`](https://wagmi.sh/react/connectors/injected#getprovider) API to target retrieve Trust Wallet when serveral wallets are installed.

It's import to note that wallets do not map directly to Connectors and are a higher-level concept. Connectors are low-level interfaces that use standards (e.g. [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193)) to link wallets to applications. In this case, Trust Wallet is served by the `InjectedConnector`.

Happy to discuss further or answer any questions you have.

**Reply by albertbinance** [reply]
Hey Tom, thanks for taking a look at our PR. 

I agree, that wagmi, as well as Trust Wallet connector, are low level, as the connectors are integrated to other libraries like rainbowkit.

Right now, using just InjectedConnector directly doesn't work well for our needs because it doesn't guarantee the order of injected providers. We believe that integrating TrustWallet into wagmi would give our extension and our users a boost.

Without our connector, we can't make sure Trust Wallet is the one being used when multiple wallets are injected. This would mean that we and others who want to use Trust Wallet would have to do extra work every time wagmi is used. 

With web3 technology advancing and more wallets becoming available (like Brave's built-in wallet), we want to make sure our users have a smooth experience connecting to our extension. 

Let us know if there's anything else we need to consider for compliance with the library.

**Reply by albertbinance** [reply]
Sending a friendly reminder on our case 🙏

**Reply by tmm** [reply]
> I agree, that wagmi, as well as Trust Wallet connector, are low level

What I'm saying above is that wagmi is the low-level framework that allows Trust Wallet to connect via the `InjectedConnector` and EIP-1993. We expose the [`InjectedConnector.getProvider`](https://wagmi.sh/react/connectors/injected#getprovider) API that folks using wagmi can use to target specific providers.

> Without our connector, we can't make sure Trust Wallet is the one being used when multiple wallets are injected.

Unfortunately this is the world that we live in for injected wallets right now. I'd encourage you time chime in on [EIP-5749](https://eips.ethereum.org/EIPS/eip-5749) so the spec might improve.
